evennia/commands/default/muxcommand.py

```diff
@@ -109,16 +123,50 @@ def parse(self):
             else:
                 args = ""
                 switches = switches[0].split('/')
+            # If user-provides switches, parse them with parser switch options.
+            if switches and self.switch_options:
+                valid_switches, unused_switches, extra_switches = [], [], []
+                for element in switches:
+                    option_check = [each for each in self.switch_options
```

Comment:
For eventual use by the end-user, it would be an idea to do 

```python
self.switch_options = [opt.lower() for opt in self.switch_options]
```
already above - this means you don't need to run `lower()` on every element again below but also that users wanting to use `switch_options` for their own ends can be sure it has already been parsed and is in lowercase.